polynote-frontend/polynote/data/result.ts

```diff
@@ -96,9 +96,9 @@ export class StackTraceElement {
 
 // maps to JVM Throwable
 // WARNING: not frozen (mutable)
-export class KernelError {
```

Comment:
I renamed this because there's a new type of `KernelStatusUpdate` called `KernelError` now.

Reply:
I'm a little confused about the different error types. In my mind, we have 3 different error scenarios:

1. Something went wrong in the Server - problem launching the kernel, or an unexpected error occurred in the internal server code (should be rare). You might need to restart the entire server to resolve the problem. 
2. Something went wrong in the Kernel - problem downloading dependencies, or an unexpected error occurred in the internal kernel code (should be rare). You might need to restart the kernel to resolve the problem. 
3. Something went wrong in user code - an exception was thrown evaluating user code. We can point to a specific cell + line in user code that triggered the error. 

Do you agree with these distinctions? I can see an argument for 1 and 2 to be combined, and there might be other cases I haven't considered. 

How would these scenarios map to the `ServerError` and `KernelError` we have now?

Reply:
`ServerError` (nee `KernelError`) is just the JavaScript-land mapping of any `Throwable` from the server. It's a data type used in a few different places and doesn't have any semantic meaning (other than that it "came from the server" somehow, and wasn't thrown by front-end code). It's used by all of the following:

1. `Error` (the message type from `Messages.scala`) is a general server-wide error, not related to any particular notebook. It's sent on the global channel.
2. `KernelError` is specifically an error that came from a particular notebook because there was an unexpected error (not an error in user code from a cell). It is specific to a notebook, but not a cell.
3.`ErrorResult` means something went wrong in the user code – either it failed to compile, or it threw an exception at runtime. It's related to a specific cell in a specific notebook (and indeed is only emitted in a `CellResult`, because it is a subtype of `Result`).

It's a little bit confusing, but I don't think needlessly so. We can clarify it with documentation comments.

Reply:
Ah I see, I was thinking of the Server/Kernel distinction from the backend perspective. 

What do you think about calling it something like `BackendError`?